kornia/geometry/transform/crop.py

```diff
@@ -135,56 +122,71 @@ def center_crop(tensor: torch.Tensor, size: Tuple[int, int]) -> torch.Tensor:
-    dst_w_half: torch.Tensor = dst_w / 2
-    src_h_half: torch.Tensor = src_h / 2
-    src_w_half: torch.Tensor = src_w / 2
-
-    start_x: torch.Tensor = src_h_half - dst_h_half
-    start_y: torch.Tensor = src_w_half - dst_w_half
+    dst_h_half = dst_h / 2
+    dst_w_half = dst_w / 2
+    src_h_half = src_h / 2
+    src_w_half = src_w / 2
 
-    end_x: torch.Tensor = start_x + dst_w - 1
-    end_y: torch.Tensor = start_y + dst_h - 1
+    start_x = src_h_half - dst_h_half
+    start_y = src_w_half - dst_w_half
 
     # [y, x] origin
     # top-left, top-right, bottom-left, bottom-right
-    points_src: torch.Tensor = torch.tensor([[
-        [start_y, start_x],
-        [start_y, end_x],
-        [end_y, start_x],
-        [end_y, end_x],
-    ]]).to(tensor.device).to(tensor.dtype)
+    points_src: torch.Tensor = _get_perspective_src(start_x, start_y, size).to(tensor.device).to(tensor.dtype)
 
     # [y, x] destination
     # top-left, top-right, bottom-left, bottom-right
-    points_dst: torch.Tensor = torch.tensor([[
-        [0, 0],
-        [0, dst_w - 1],
-        [dst_h - 1, 0],
-        [dst_h - 1, dst_w - 1],
-    ]]).to(tensor.device).to(tensor.dtype)
+    points_dst: torch.Tensor = _get_perspective_dst(size).to(tensor.device).to(tensor.dtype)
 
     # warping needs data in the shape of BCHW
     is_unbatched: bool = tensor.ndimension() == 3
     if is_unbatched:
         tensor = torch.unsqueeze(tensor, dim=0)
 
-    # compute transformation between points and warp
-    dst_trans_src: torch.Tensor = get_perspective_transform(
-        points_src, points_dst)
-    dst_trans_src = dst_trans_src.repeat(tensor.shape[0], 1, 1)
-
-    patches: torch.Tensor = warp_perspective(
-        tensor, dst_trans_src, (int(dst_h), int(dst_w)))
+    patches = _crop_wrapper(tensor, points_src, points_dst)
 
     # return in the original shape
     if is_unbatched:
         patches = torch.squeeze(patches, dim=0)
 
     return patches
+
+
+def _get_perspective_src(start_x: int, start_y: int, size: Tuple[int, int]) -> torch.Tensor:
+
+    end_x = start_x + size[1] - 1
+    end_y = start_y + size[0] - 1
+
+    # [y, x] origin
+    # top-left, top-right, bottom-left, bottom-right
+    return torch.tensor([[
+        [start_y, start_x],
+        [start_y, end_x],
+        [end_y, start_x],
+        [end_y, end_x],
+    ]])
+
+
+def _get_perspective_dst(size: Tuple[int, int]) -> torch.Tensor:
+    # [y, x] destination
+    # top-left, top-right, bottom-left, bottom-right
+    return torch.tensor([[
+        [0, 0],
+        [size[1] - 1, 0],
+        [size[1] - 1, size[0] - 1],
+        [0, size[0] - 1],
+    ]])
+
+
+def _crop_wrapper(input: torch.Tensor, src: torch.Tensor, dst: torch.Tensor) -> torch.Tensor:
+    dst_trans_src: torch.Tensor = get_perspective_transform(src, dst)
+    dst_trans_src = dst_trans_src.repeat(input.shape[0], 1, 1)
+
+    size = (int(dst.squeeze()[-1][-1] + 1), int(dst.squeeze()[1][0] + 1))
```

Comment:
could you add a comment explaining what's this ?  simplify or make tmp variable names